tests/lib/external/snapd-testing-tools/utils/log-analyzer

```diff
@@ -1,330 +1,220 @@
-        exit 1
-    fi
-    exec_exp="$(echo "$exec_exp" | tr ',' ' ')"
-    _check_log "$log"
-
-    local all_tasks executed_tasks
-    all_tasks="$(list_all_tasks "$exec_exp")"
-    executed_tasks="$(jq -r '.[] | select( .type == "phase") | select( .verb == "Executing") | .task' "$log")"
-    _intersection_tasks_lists "$executed_tasks" "$all_tasks"
-}
-
-list_aborted_tasks() {
-    local exec_exp="$1"
-    local log="$2"
-
-    if [ -z "$exec_exp" ]; then
-        echo "log.analyzer: execution expression for spread cannot be empty"
-        exit 1
-    fi
-    exec_exp="$(echo "$exec_exp" | tr ',' ' ')"
-    _check_log "$log"
-
-    local all_tasks executed_tasks failed_tasks_prepare failed_tasks_restore failed_tasks
-    all_tasks="$(list_all_tasks "$exec_exp")"
-    executed_tasks="$(list_executed_tasks "$exec_exp" "$log")"
-    failed_tasks="$(list_failed_tasks "$exec_exp" "$log")"
-    failed_tasks_prepare="$(_list_failed task prepare "$log")"
-    failed_tasks_restore="$(_list_failed task restore "$log")"
-
-    # In case no tasks for the expression, the aborted list is empty
-    if [ -z "$all_tasks" ]; then
-        return
-    fi
-
-    # In case no tasks are successfully executed, all the tasks - the failed ones are the aborted
-    if [ -z "$executed_tasks" ]; then
-        exec_and_failed_tasks="$(_list_executed_and_failed_tasks "$exec_exp" "$log")"
-        _diff_tasks_lists "$all_tasks" "$exec_and_failed_tasks"
-        return
-    fi
-    
-    # In other cases the aborted tasks are all the tasks - the executed - the that failed 
-    _diff_tasks_lists "$all_tasks" "$executed_tasks"
-}
-
-main() {
-    if [ $# -eq 0 ]; then
-        show_help
-        exit 0
-    fi
-
-    local subcommand="$1"
-    local action=
-    while [ $# -gt 0 ]; do
-        case "$1" in
-            -h|--help)
-                show_help
-                exit 0
-                ;;
-            *)
-                action=$(echo "$subcommand" | tr '-' '_')
-                shift
-                break
-                ;;
-        esac
-    done
-
-    if [ -z "$(declare -f "$action")" ]; then
-        echo "log.analyzer: no such command: $subcommand" >&2
-        show_help
-        exit 1
-    fi
-
-    "$action" "$@"
-}
-
-main "$@"
+#!/usr/bin/env python3
+
+# The following script requires Python 3.9 or higher
+
+import argparse
+import json
+import subprocess
+from typing import Callable, TypedDict, Literal, Union
+
+
+class SpreadLog_TypePhase(TypedDict):
+    type: Literal["phase"]
+    task: str
+    verb: str
+
+
+class SpreadLogDetail(TypedDict):
+    lines: list[str]
+
+
+class SpreadLog_TypeResult(TypedDict):
+    type: Literal["result"]
+    result_type: str
+    level: str
+    stage: str
+    detail: SpreadLogDetail
+
+
+SpreadLog = Union[SpreadLog_TypePhase, SpreadLog_TypeResult]
+
+
+def filter_with_spread(exec_param: str) -> list[str]:
+    exec_param = exec_param.replace(",", " ")
```

Comment:
Should we instead define the type of the argument as `string` with `nargs="+"` and remove this replacement and the split below?